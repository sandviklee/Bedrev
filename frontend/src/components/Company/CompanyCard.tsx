import { Company } from "../../types/types";
import usersIcon from "/icons/users.svg";
import briefcaseIcon from "/icons/briefcase.svg";
import globeIcon from "/icons/globe.svg";

const CompanyCard = (company: Company) => {
  return (
    <div className="w-full flex flex-row rounded-md overflow-hidden">
      <div className="w-full px-[2%] py-[2%] flex-row bg-purple-lighter border-purple border-2 rounded-r-md">
        <p className="sm:leading-[40px] leading-[25px] sm:text-lg text-[12px] sm:font-medium font-bold font-whyte mb-[-6px]">
          {company.navn.toUpperCase()}
        </p>
        <p className="truncate sm:leading-[30px] sm:text-sm text-[12px] font-inter max-w-[700px] h-[70px]">
          {company.vedtektsfestetformaal}
        </p>
        <div className="hidden lg:flex flex-row gap-[5%] mt-2">
          <div className="flex flex-row gap-2 items-center">
            <img alt="employees" src={usersIcon} />
            {company.antallansatte}
          </div>
          <div className="flex flex-row gap-2 items-center">
            <img alt="employees" src={briefcaseIcon} />
            {"Bedrift"}
          </div>
          <div className="flex flex-row gap-2 items-center">
            <img alt="employees" src={globeIcon} />
            {company.postadresse_kommune}
          </div>
        </div>
      </div>
    </div>
  );
};

export default CompanyCard;
