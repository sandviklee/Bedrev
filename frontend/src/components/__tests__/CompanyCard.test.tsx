import { test, describe, expect } from "vitest";
import CompanyCard from "../Company/CompanyCard.tsx";
import { render, screen } from "@testing-library/react";

describe("CompanyCard component", () => {
  const mockCompany = {
    navn: "Bedrift",
    vedtektsfestetformaal: "Lage ting",
    organisasjonsnummer: 123456789,
    antallansatte: 100,
    postadresse_kommune: "Oslo",
  };

  test("renders correctly", () => {
    render(<CompanyCard {...mockCompany} />);
    expect(screen.getByText("Bedrift")).toBeInTheDocument();
    expect(screen.getByText("Lage ting")).toBeInTheDocument();
    expect(screen.getByText("100")).toBeInTheDocument();
    expect(screen.getByText("Oslo")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<CompanyCard {...mockCompany} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
