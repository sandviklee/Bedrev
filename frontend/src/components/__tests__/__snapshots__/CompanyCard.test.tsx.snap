// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full flex flex-row rounded-md overflow-hidden"
  >
    <div
      class="w-full px-[2%] py-[2%] flex-row bg-purple-lighter border-purple border-2 rounded-r-md"
    >
      <p
        class="sm:leading-[40px] leading-[25px] sm:text-lg text-[12px] sm:font-medium font-bold font-whyte mb-[-6px]"
      >
        BEDRIFT
      </p>
      <p
        class="truncate sm:leading-[30px] sm:text-sm text-[12px] font-inter max-w-[700px] h-[70px]"
      >
        Lage ting
      </p>
      <div
        class="hidden lg:flex flex-row gap-[5%] mt-2"
      >
        <div
          class="flex flex-row gap-2 items-center"
        >
          <img
            alt="employees"
            src="/project2/icons/users.svg"
          />
          100
        </div>
        <div
          class="flex flex-row gap-2 items-center"
        >
          <img
            alt="employees"
            src="/project2/icons/briefcase.svg"
          />
          Bedrift
        </div>
        <div
          class="flex flex-row gap-2 items-center"
        >
          <img
            alt="employees"
            src="/project2/icons/globe.svg"
          />
          Oslo
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
